lib/pages/settings/view/settings_page.dart

```diff
@@ -18,6 +24,15 @@ class SettingsPage extends StatefulWidget {
 }
 
 class SettingsPageState extends State<SettingsPage> {
+  User user;
+  AuthProvider authProvider;
+
+  @override
+  void initState() {
+    super.initState();
+    authProvider = Provider.of(context, listen: false);
+  }
```

Comment:
This is unnecessary, you can just get the `authProvider` in the `build` method and don't need to pass `listen: false`.